src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -499,6 +500,7 @@ private static Details getDetailsForLoops(DetailAST ast) {
             final DetailAST lcurly;
             final DetailAST nextToken;
             final int tokenType = ast.getType();
+            boolean shouldCheckLastRcurly = false;
```

Comment:
make boolean `final` and move `false` assignment to inside the `if`.
This way it follows the patterns of the other `final` variables.